### PR TITLE
Fix/empty transactions query

### LINKED ISSUE #174: Postgres error when querying for transactions
Querying for transactions using a newly derived address (path index `1`) returns this error:

```console
$ curl -X 'GET' \
  'http://127.0.0.1:9090/addresses/1B9DJTNbpPB17qFgh3MJm4hVJ6iFnraDJAGAYYxndfc8o/transactions' \
  -H 'accept: application/json'

Internal server error%
```

Logs:
```
[info] 2022-03-21 11:01:20,437 [-dispatcher-7] ERROR a.a.ActorSystemImpl                    - Exception when handling request by: GET /addresses /[address] /transactions ?page(>=1) ?limit(all(>=0,<=100)) ?reverse -> one of({body as application/json (UTF-8)}|{body as application/json (UTF-8)}|{body as application/json (UTF-8)}|{body as application/json (UTF-8)}|{body as application/json (UTF-8)})/{body as application/json (UTF-8)}
[info] java.lang.RuntimeException: org.postgresql.util.PSQLException: ERROR: syntax error at or near ")"
```

Full traceback is this: https://gist.github.com/nop33/e72651a4b58335ed7686416ab3f255bb

Note: When querying the transactions of the address with path index `0`, there is no error:
```
curl -X 'GET' \
  'http://127.0.0.1:9090/addresses/1J1h1fmDX5sfiubXnMFSzKQemV7DQX2Vzxsj3jL1GkzVk/transactions' \
  -H 'accept: application/json'

[{"hash":"1615d384589b3bddf4d38c9fc9f25094bf2237ad7864a68e795d50546a70b3b8","blockHash":"f287969c45ff71c20c1c9e479be816a027bc4b97c6b9b090f171907253d91e25","timestamp":1231006505000,"inputs":[],"outputs":[{"hint":1819796851,"key":"e9b755a614161789597aa523c6922ece9eaa84f93a00348d2b4f96f8b52d3be2","amount":"100000000000000000000000","address":"1J1h1fmDX5sfiubXnMFSzKQemV7DQX2Vzxsj3jL1GkzVk","lockTime":1636379973000},{"hint":1819796851,"key":"1cafc795fe70b76dcb1b4669aacdecead63179ae2779bc91234c1eac2e57538b","amount":"10000000000000000000","address":"1J1h1fmDX5sfiubXnMFSzKQemV7DQX2Vzxsj3jL1GkzVk","lockTime":1722693573000}],"gasAmount":20000,"gasPrice":"100000000000"}]%
```

Steps I followed to run explorer-backend:

```console
$ sbt --version
sbt version in this project: 	1.6.2
sbt script version: 1.6.2

$ psql --version
psql (PostgreSQL) 14.2

$ java --version
openjdk 17 2021-09-14
OpenJDK Runtime Environment (build 17+35-2724)
OpenJDK 64-Bit Server VM (build 17+35-2724, mixed mode, sharing)

$ git show
commit 505566aabdc655968bfaa640a6c3810adcb75a1b (HEAD -> v1.6.0-rc2, tag: v1.6.0-rc2)

$ psql postgres
postgres=# drop database explorer; create database explorer;

$ sbt clean
[info] welcome to sbt 1.6.2 (Oracle Corporation Java 17)
[info] loading settings for project explorer-backend-build from plugins.sbt ...
[info] loading project definition from /Users/ilias/dev/alephium/explorer-backend/project
[info] loading settings for project root from build.sbt ...
[info] set current project to explorer-backend (in build file:/Users/ilias/dev/alephium/explorer-backend/)
[success] Total time: 1 s, completed 21 Mar 2022, 10:58:55

$ BLOCKFLOW_CHAIN_ID=1 sbt app/run
[info] welcome to sbt 1.6.2 (Oracle Corporation Java 17)
[info] loading settings for project explorer-backend-build from plugins.sbt ...
[info] loading project definition from /Users/ilias/dev/alephium/explorer-backend/project
[info] loading settings for project root from build.sbt ...
[info] set current project to explorer-backend (in build file:/Users/ilias/dev/alephium/explorer-backend/)
[info] compiling 99 Scala sources to /Users/ilias/dev/alephium/explorer-backend/app/target/scala-2.13/classes ...
[info] running (fork) org.alephium.explorer.Main
[info] 2022-03-21 11:00:27,309 [main]          INFO  o.a.explorer.Main$                     - Starting Application
[info] 2022-03-21 11:00:27,353 [main]          INFO  o.a.explorer.Main$                     - Build info: name: explorer-backend-app, scalaVersion: 2.13.3, sbtVersion: 1.6.2, commitId: 505566aabdc655968bfaa640a6c3810adcb75a1b, branch: v1.6.0-rc2, releaseVersion: 1.6.0-rc2
[info] 2022-03-21 11:00:27,970 [-dispatcher-4] INFO  a.e.s.Slf4jLogger                      - Slf4jLogger started
[info] 2022-03-21 11:00:30,115 [main]          INFO  c.z.h.HikariDataSource                 - db.db - Starting...
[info] 2022-03-21 11:00:30,134 [main]          INFO  c.z.h.HikariDataSource                 - db.db - Start completed.
[info] 2022-03-21 11:00:30,238 [main]          DEBUG i.m.c.u.i.l.InternalLoggerFactory      - Using SLF4J as the default logging framework
[info] 2022-03-21 11:00:32,703 [-dispatcher-5] DEBUG o.a.e.s.MempoolSyncService$Impl        - Syncing mempol
[info] 2022-03-21 11:00:32,821 [-dispatcher-5] DEBUG o.a.e.s.TokenSupplyService$Impl        - Computing token supply
[info] 2022-03-21 11:00:33,363 [-dispatcher-5] DEBUG o.a.e.s.HashrateService$Impl           - Updating hashrates
[info] 2022-03-21 11:00:33,475 [dispatcher-12] DEBUG o.a.e.s.TokenSupplyService$Impl        - Token supply updated
[info] 2022-03-21 11:00:33,794 [-dispatcher-8] DEBUG o.a.e.s.HashrateService$Impl           - Hashrates updated in 429 ms
[info] 2022-03-21 11:00:33,861 [dispatcher-12] DEBUG o.a.e.s.MempoolSyncService$Impl        - Mempool synced
[info] 2022-03-21 11:00:34,526 [-dispatcher-5] INFO  o.a.e.Application                      - Listening http request on ServerBinding(/127.0.0.1:9090)
[info] 2022-03-21 11:00:38,870 [Timer-1]       DEBUG o.a.e.s.MempoolSyncService$Impl        - Syncing mempol
[info] 2022-03-21 11:00:38,999 [-dispatcher-5] DEBUG o.a.e.s.MempoolSyncService$Impl        - Mempool synced
```

### PR DESCRIPTION
**Pull request by tdroxler**
Resolves #174

### CODE REVIEW

File: app/src/main/scala/org/alephium/explorer/persistence/queries/InputQueries.scala [hunk 1]
```diff
@@ -37,9 +37,10 @@ object InputQueries extends CustomTypes {
 
   /** Inserts inputs or ignore rows with primary key conflict */
   def insertInputs(inputs: Iterable[InputEntity]): DBActionW[Int] =
-    QuerySplitter.splitUpdates(rows = inputs, columnsPerRow = 9) { (inputs, placeholder) =>
-      val query =
-        s"""
+    if (inputs.nonEmpty) {
+      QuerySplitter.splitUpdates(rows = inputs, columnsPerRow = 9) { (inputs, placeholder) =>
```
Comment (simerplaha) [inline app/src/main/scala/org/alephium/explorer/persistence/queries/InputQueries.scala:41]: `QuerySplitter.splitUpdates` already checks for empty inputs [here](https://github.com/alephium/explorer-backend/blob/master/app/src/main/scala/org/alephium/explorer/persistence/queries/QuerySplitter.scala#L123). The [test-case](https://github.com/alephium/explorer-backend/blob/master/app/src/test/scala/org/alephium/explorer/persistence/queries/QuerySplitterSpec.scala#L66). So this check `if (inputs.nonEmpty)` is probably not needed?

**Review by polarker — APPROVED** [review]
LGTM. Not sure if there will be formatting issues.

File: app/src/main/scala/org/alephium/explorer/persistence/queries/OutputQueries.scala [hunk 2]
```diff
@@ -35,9 +35,10 @@ object OutputQueries extends CustomTypes {
   /** Inserts outputs or ignore rows with primary key conflict */
   // scalastyle:off magic.number
   def insertOutputs(outputs: Iterable[OutputEntity]): DBActionW[Int] =
-    QuerySplitter.splitUpdates(rows = outputs, columnsPerRow = 11) { (outputs, placeholder) =>
-      val query =
-        s"""
+    if (outputs.nonEmpty) {
+      QuerySplitter.splitUpdates(rows = outputs, columnsPerRow = 11) { (outputs, placeholder) =>
```
Comment (simerplaha) [inline app/src/main/scala/org/alephium/explorer/persistence/queries/OutputQueries.scala:39]: Same as above. Emptiness is already checked by `QuerySplitter`.

File: app/src/main/scala/org/alephium/explorer/persistence/queries/InputQueries.scala (hunk 1 above)
Comment (tdroxler) [reply app/src/main/scala/org/alephium/explorer/persistence/queries/InputQueries.scala:41]: oh great, I miss that :+1: I will update

Comment (tdroxler) [reply app/src/main/scala/org/alephium/explorer/persistence/queries/InputQueries.scala]: I just removed the last commit

**Review by simerplaha — APPROVED** [review]
👍🏼 🚀

**Pull request merged**